pkg/controllers/nodeclass/validation.go

```diff
@@ -149,81 +159,97 @@ func (v *Validation) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass)
 	}
 
 	if v.dryRunDisabled {
-		nodeClass.StatusConditions().SetTrueWithReason(
-			v1.ConditionTypeValidationSucceeded,
-			ConditionReasonDryRunDisabled,
-			"Dry run is disabled",
-		)
+		nodeClass.StatusConditions().SetTrue(v1.ConditionTypeValidationSucceeded)
 		v.cache.SetDefault(v.cacheKey(nodeClass, tags), "")
 		return reconcile.Result{}, nil
 	}
 
-	for _, isValid := range []validatorFunc{
-		v.validateCreateFleetAuthorization,
-		v.validateCreateLaunchTemplateAuthorization,
-		v.validateRunInstancesAuthorization,
-	} {
-		if failureReason, requeue, err := isValid(ctx, nodeClass, nodeClaim, tags); err != nil {
-			return reconcile.Result{}, err
-		} else if requeue {
-			return reconcile.Result{Requeue: true}, nil
-		} else if failureReason != "" {
-			v.cache.SetDefault(v.cacheKey(nodeClass, tags), failureReason)
-			nodeClass.StatusConditions().SetFalse(
-				v1.ConditionTypeValidationSucceeded,
-				failureReason,
-				ValidationConditionMessages[failureReason],
-			)
-			return reconcile.Result{}, nil
-		}
+	failureReason, requeue, launchTemplate, err := v.validateCreateLaunchTemplateAuthorization(ctx, nodeClass, nodeClaim, tags)
+	if reconcileResult, shouldReturn, err := v.handleValidationResult(failureReason, requeue, err, nodeClass, tags); shouldReturn {
+		return reconcileResult, err
+	}
+
+	failureReason, requeue, err = v.validateCreateFleetAuthorization(ctx, nodeClass, nodeClaim, tags, launchTemplate)
+	if reconcileResult, shouldReturn, err := v.handleValidationResult(failureReason, requeue, err, nodeClass, tags); shouldReturn {
+		return reconcileResult, err
+	}
+
+	failureReason, requeue, err = v.validateRunInstancesAuthorization(ctx, nodeClass, nodeClaim, tags, launchTemplate)
+	if reconcileResult, shouldReturn, err := v.handleValidationResult(failureReason, requeue, err, nodeClass, tags); shouldReturn {
+		return reconcileResult, err
 	}
 
 	v.cache.SetDefault(v.cacheKey(nodeClass, tags), "")
 	nodeClass.StatusConditions().SetTrue(v1.ConditionTypeValidationSucceeded)
 	return reconcile.Result{}, nil
 }
 
-type validatorFunc func(context.Context, *v1.EC2NodeClass, *karpv1.NodeClaim, map[string]string) (string, bool, error)
-
-func (v *Validation) validateCreateFleetAuthorization(
-	ctx context.Context,
+func (v *Validation) handleValidationResult(
```

Comment:
This function should also requeueAfter: 10m if we are planning to requeue consistently -- we might be able to factor this slightly different if we return RequeueAfter, actually. Returning a return argument like `shouldReturn` feels a little off. What if we can check if the result is empty? If it's empty, then we just move on to the next function -- the only ways to return early for this function are to set a RequeueAfter or return an error?